cm-server/src/main/java/ro/msg/cm/controller/UpdateStudyYearController.java

```diff
@@ -0,0 +1,178 @@
+import java.util.*;
+
+import static java.util.Calendar.YEAR;
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo;
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.methodOn;
+
+@RestController
+@RequestMapping("/api")
+public class UpdateStudyYearController {
+    private final CandidateRepository candidateRepository;
+    private final StartYearProperties startYearProperties;
+
+    @Autowired
+    public UpdateStudyYearController(CandidateRepository candidateRepository, StartYearProperties startYearProperties) {
+        this.candidateRepository = candidateRepository;
+        this.startYearProperties = startYearProperties;
+    }
+
+    @RequestMapping(value = "/candidates", method = RequestMethod.GET, produces = "application/hal+json")
+    public Resources updateUniversityYear() throws ParseException {
+
+        Iterable<Candidate> candidates = findAll();
+        List<Candidate> candidateList = new ArrayList<>();
+        candidates.forEach(candidateList::add);
+        List<Resource<Candidate>> resources = new LinkedList<>();
+        for (Candidate candidate : candidates) {
+            candidate.setCurrentStudyYear(determineYearBasedOnDuration(candidate));
+            resources = candidateToResource(candidateList);
+        }
+        return new Resources<>(resources);
+
+    }
+
+    private Resource<Candidate> getResource(Candidate candidate) throws ParseException {
+
+        Resource<Candidate> resource = new Resource<>(candidate);
+        resource.add(linkTo(methodOn(UpdateStudyYearController.class).getResourceCandidateById(candidate.getId())).withSelfRel());
+        resource.add(linkTo(methodOn(UpdateStudyYearController.class).getResourceCandidateById(candidate.getId())).withRel("candidate"));
+        resource.add(linkTo(methodOn(CandidateSkillsController.class).getCandidateSkills(candidate.getId())).withRel("candidateSkillsList"));
+        resource.add(linkTo(methodOn(EducationController.class).getEducation(candidate.getId())).withRel("education"));
+        resource.add(linkTo(methodOn(CandidateNotesController.class).getCandidateNotes(candidate.getId())).withRel("candidateNotesList"));
+
+        return resource;
+    }
+
+    private List<Resource<Candidate>> candidateToResource(List<Candidate> candidates) throws ParseException {
+        List<Resource<Candidate>> resources = new ArrayList<>(candidates.size());
+        for (Candidate candidate : candidates) {
+            Resource<Candidate> resource = getResource(candidate);
+            resources.add(resource);
+        }
+        return resources;
+    }
+
+    @RequestMapping(value = "candidates/{id}", method = RequestMethod.GET, produces = "application/hal+json")
+    public Resource getResourceCandidateById(@PathVariable(value = "id") Long id) throws ParseException {
+
+        Candidate candidate = candidateRepository.findOne(id);
+        Resource<Candidate> resource = new Resource<>(candidate);
```

Comment:
This should also use the `getResource()` method such that the resource obtained from reading a single candidate is the same as an element of the resources obtained when reading all the candidates.